glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/PingUploadWorker.kt

```diff
@@ -192,10 +99,27 @@ class PingUploadWorker(context: Context, params: WorkerParameters) : Worker(cont
      * @return The [androidx.work.ListenableWorker.Result] of the computation
      */
     override fun doWork(): Result {
-        return when {
-            !Glean.getUploadEnabled() -> Result.failure()
-            !uploadPings() -> Result.retry()
-            else -> Result.success()
-        }
+        do {
+            val incomingTask = LibGleanFFI.INSTANCE.glean_get_upload_task()
+            when (val action = incomingTask.toPingUploadTask()) {
+                is PingUploadTask.Upload -> {
+                    // Upload the ping request.
+                    // If the status is `null` there was some kind of unrecoverable error
+                    // so we return a known unrecoverable error status code
+                    // which will ensure this gets treated as such.
+                    val status = Glean.httpClient.doUpload(
+                            action.request.path,
+                            action.request.body,
+                            action.request.headers,
+                            Glean.configuration
+                    ) ?: UNRECOVERABLE_ERROR_STATUS_CODE
```

Comment:
So ... adressing that and the previous comment: I came up with a solution: Instead of bool or `Int?` we introduce a new type we can _properly_ match on.
See https://github.com/mozilla/glean/pull/813/commits/c68c2ec6242d14723bef042f29d3adc997ac2be1

This **is** a breaking change, becuase the PingUploader interface is public and implemented in android-components. The fix in a-c is simple.

I am still converting it to "HTPP status codes" on the way into Rust, but we could address that in a followup.

It's more code than just flipping around some minor types, but IMO it's easier to understand.